src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs

```diff
@@ -227,6 +227,8 @@ private static ReportDiagnostic MapSeverityToReport(DiagnosticSeverity severity)
                     return ReportDiagnostic.Warn;
                 case DiagnosticSeverity.Error:
                     return ReportDiagnostic.Error;
+                case DiagnosticSeverity.Suggestion:
```

Comment:
tightly coupling each severity type to where it shows up seems making system a bit too much inflexible. in this design, we need new severity on every combination on where things show up. right now we have 2 (error list and editor) so we need 4 severity at least ((error/warning)/info/suggestion/hidden) which represent (on/on, on/off, off/on, off/off) respectively.

as soon as we have 1 more such place, we need 8 severity. or do something like unnecessary custom tag. which, when put together, feels like we have 2 different system for same thing.

I think visual presentation (where it shows up and how it shows up) is options for diagnostics which I believe should be just flags. now complexity is linear rather than exponential. (each flag is independent to others rather than we need all combinations)